cpp/src/arrow/json/rapidjson_defs.h

```diff
@@ -36,10 +36,6 @@
 #include "arrow/util/sse_util.h"
 
 // enable SIMD whitespace skipping, if available
-#if defined(ARROW_HAVE_SSE2)
-#define RAPIDJSON_SSE2 1
```

Comment:
We should keep defining `RAPIDJSON_SSE2`, no?

Reply:
From below rapidjson code, looks RAPIDJSON_SSE2 should be kept (though I suspect that code should check SSE42 first). Will update this patch to include RAPIDJSON_SSE2.
https://github.com/Tencent/rapidjson/blob/master/test/perftest/rapidjsontest.cpp#L30